File: Easy/Java/Third_Maximum_Number(414).java
class Solution {
    public int thirdMax(int[] nums) {
        long max = Long.MIN_VALUE;
        long secondMax = max, thirdMax = max;
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > max) {
                thirdMax = secondMax;
                secondMax = max;
                max = nums[i];
            } else if (nums[i] > secondMax && nums[i] != max) {
                thirdMax = secondMax;
                secondMax = nums[i];
            } else if (nums[i] > thirdMax && nums[i] != max && nums[i] != secondMax) {
                thirdMax = nums[i];
            }
        }
            
        return (int) (thirdMax == Long.MIN_VALUE ? max : thirdMax);
    }
}

File: Easy/C++/Third_Maximum_Number(414).cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        int i,k=0,c=0,flag;
        sort(nums.begin(),nums.end());
        for(i=nums.size()-2;i>=0;i--)
        {
            if(nums.at(i)<nums.at(i+1))
                c++;
            if(c==2)
            {
                flag=nums.at(i);
                k=-1;
                break;
            }
        }
        if(k==-1)
            return flag;
        else
            return nums.at(nums.size()-1);
    }
};